### PR TITLE
[RTL] Change direction of content in forms at runtime

### PR DESCRIPTION
**Pull request by devgurjar**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](http://opensource.adobe.com/cla.html).
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes and the overall coverage did not decrease.
- [ ] All unit tests pass on CircleCi.
- [ ] I ran all tests locally and they pass.

### CODE REVIEW

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/site/js/formcontainerview.js [hunk 1]
```diff
@@ -83,6 +83,10 @@
             // before initializing the form container load all the locale specific json resources
             // for runtime
             const formLanguage = formContainer.getLang();
+            if (formLanguage) {
```
Comment (sakshi-arora1) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/site/js/formcontainerview.js:86]: before setting it check if "dir" is already set at body

**Review by sakshi-arora1 — APPROVED** [review]

**Comment by devgurjar** [comment]
One test is failing in CI, but passing locally.

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java [hunk 2]
```diff
@@ -259,6 +259,30 @@ public String getLang() {
...
+        if (isLangRTL()) {
+            return "rtl";
+        } else {
+            return "ltr";
+        }
+    }
+
+    private boolean isLangRTL() {
+        String language = getLang();
+        if (StringUtils.isNotEmpty(language)) {
```
Comment (rismehta) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:273]: String language = getLang();
return Arrays.stream(new String[] {"ar", "iw", "he", "fa", "ur", "ug", "ji", "yi"})
    .anyMatch(lang -> lang.equalsIgnoreCase(language.split("-")[0]));

File: bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java [hunk 3]
```diff
@@ -259,6 +259,30 @@ public String getLang() {
         }
     }
 
+    @Override
+    public String getDirection() {
+        if (isLangRTL()) {
```
Comment (rismehta) [inline bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java:264]: Can you create something in ComponentUtils for this, 

`ComponentUtils.getLanguageDirection` and move the entire logic there

**Review by rismehta — CHANGES_REQUESTED** [review]
Please don't raise a PR without unit test cases. Thanks.

File: ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/site/js/formcontainerview.js [hunk 4]
```diff
@@ -83,6 +83,9 @@
             // before initializing the form container load all the locale specific json resources
             // for runtime
             const formLanguage = formContainer.getLang();
+//            if (formContainer._element && !formContainer._element.dir && formLanguage) {
```
Comment (rismehta) [inline ui.af.apps/src/main/content/jcr_root/apps/core/fd/components/form/container/v2/container/clientlibs/site/js/formcontainerview.js:86]: Please remove this, not sure why is it commented, let's keep the code source clean

**Comment by devgurjar** [comment]
Closing this PR

**Pull request closed without merging**